teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java

```diff
@@ -288,6 +288,17 @@ private void validateOrDefaultEpoch() {
               "The specified epoch %s is greater than current epoch %s, cannot continue.",
               epoch, maybeEpoch.get()));
       System.exit(1);
+    } else if (maybeEpoch.isPresent()
+        && spec.computeMinimumViableEpoch(spec.computeStartSlotAtEpoch(maybeEpoch.get()))
+            .isGreaterThan(epoch)) {
+      SUB_COMMAND_LOG.error(
+          String.format(
+              "The specified epoch %s (%s) is on a different milestone to the chains current epoch %s (%s), cannot continue.",
+              epoch,
+              spec.atEpoch(epoch).getMilestone(),
+              maybeEpoch.get(),
+              spec.atEpoch(maybeEpoch.get()).getMilestone()));
+      System.exit(1);
```

Comment:
We shouldn't need this here because the REST API should reject the exit for us. If not we need to fix the REST API.